### DISCUSSION TITLE
en/lcci/1.1/

### DISCUSSION THREAD

**Discussion by giscus** [doocs/leetcode#3737] [Announcements]
# en/lcci/1.1/

LeetCode solutions in any programming language

https://leetcode.doocs.org/en/lcci/1.1/

**Comment by jumael99** [comment]
Hi! Without bit manipulation can you give some other easy solution?

**Reply by devmustafa03** [reply]
const isCheckingSimilarCharacter = (word) => {
    const currentWord = new Set();
    
    for (let char of word) {
        if (currentWord.has(char)) {
            return false;
        }
        currentWord.add(char);
    }
    
    return true;
};

This is for javascript

**Comment by rajputanisha** [comment]
#include <bits/stdc++.h>
using namespace std;
bool fun(string s,int n ){
 unordered_map<char,int>mp;
 for(auto it :s){
   mp[it]++;
 }
 for(auto it :mp){
   if(it.second>1){
     return false;
   }

 }
 return true;
}
int main() 
{
string s = "lswtcde";
    int n = s.size();
    bool ans = fun(s, n);
    cout << ans << endl;
    
    return 0;
}

**Reply by rajputanisha** [reply]
In this we use map to store the key value pair. If count greater than one means there are duplicate present . we simply return false. otherwise we will simply return true if count of all elements are equal to one. i e all elements are unique
map store element like this for example  string example = "Leetcode" 
L ->1
e ->2
t ->1
c ->1
o ->1
d ->1
e ->1
this is simple representation of unordered_map

**Comment by ADITYAPIMPALE11127** [comment]
import java.util.HashSet;
import java.util.Scanner;
public class UniqueString {
    public static boolean UniqueChars(String str){
        HashSet<Character> seen = new HashSet<>();
        for(int i = 0; i < str.length();i++){
            char  ch = str.charAt(i);
            if(seen.contains(ch)){
                return false;
            } seen.add(ch);
        }
        return true;
    }
    public static void main(String[] args){
        Scanner input  = new Scanner(System.in);
        String str1 = input.nextLine();

        System.out.print(str1+" and the string is "+UniqueChars(str1));
    }
}
